2024FALL-SWPP/team-project-for-2024-fall-swpp-team-16
# Feat/Inspector Note add color when played

## Description
Refactored `DoorController.cs` to improve maintainability and added assets so that each played note changes color to the appropriate color (1 : red, 2: orange, yellow, green, blue, purple, pink). 
## Related Issue
[Fix] 악보 UI 자연스럽게 (하얀박스가 아니게) 바꾸기 #45

## Changes Made
- [ ] Added assets in `04.Images/Sprites/UI`: white version of original notes so that notes can be recolored in Unity. Assets are named `4_w`, `4d_w`, `6_w`, `8_w`, `8d_w`
- [ ] hard coded notes colors and replaced `SpawnedColorNote` with `UpdateColoredNote`. For each note, only 1 note object is necessary (no need to spawn) as that object's color is directly changed.
``` 
    private Color GetNoteColor(int note)
    {
        switch (note)
        {
            case 1: return Color.red;
            case 2: return new Color(1f, 0.5f, 0f); // Orange
            case 3: return Color.yellow;
            case 4: return Color.green;
            case 5: return Color.blue;
            case 6: return new Color(0.5f, 0f, 0.5f); // Purple
            case 7: return new Color(1f, 0.75f, 0.8f); // Pink
            default: return Color.white;
        }
    } 
```
```
 private void UpdateColoredNote(int note)
    {
        if (playedNotes.Count - 1 < coloredNotes.Length)
        {
            GameObject noteObject = coloredNotes[playedNotes.Count - 1];
            Image image = noteObject.GetComponent<Image>();
            if (image != null)
            {
                image.color = GetNoteColor(note);

                // Create ripple effect
                GameObject ripple = Instantiate(rippleEffectPrefab, noteObject.transform.position + new Vector3(0, 0, -1), Quaternion.identity);
                ripple.transform.SetParent(noteObject.transform);
                ripple.transform.localScale = Vector3.one * 10;
                Destroy(ripple, 1.0f);

               // StartCoroutine(ScaleEffect(noteObject.transform)); // IF NOTE SHOULD SCALE WHEN PLAYED UNCOMMENT THIS
            }
        }
    }
```
- [ ] limited scaling note after playing to ripple effect because it looked unnatural(can reverse by uncommenting line 98)
```
   // StartCoroutine(ScaleEffect(noteObject.transform)); // IF NOTE SHOULD SCALE WHEN PLAYED UNCOMMENT THIS
```


## How to Test This Branch
In Hierarchy, Canvas/ MusicPaper/ Notes, freely add or remove notes (UI Image). To add, Add a new UI Image and select between `4_w`, `4d_w`, `6_w`, `8_w`, `8d_w` as source image. Then, in hierarchy in Door/jj_door_1_white, adjust the number of AnswerNotes (and assign note value (do = 1, re = 2, mi =3)) and add the note object to ColoredNotes list. The index of the answernotes (what music note should be played) and colorednotes (which object in the hierarchy is it) should correspond.

## Screenshots (if applicable)


https://github.com/user-attachments/assets/63b910dd-ccb7-4102-a296-de59bcca24f4
